src/Autosuggest.js

```diff
@@ -136,6 +136,8 @@ export default class Autosuggest extends Component {
       if (this.willRenderSuggestions(nextProps)) {
         if (this.state.isCollapsed && !this.justSelectedSuggestion) {
           this.revealSuggestions();
+        } else if (!nextProps.highlightFirstSuggestion) {
+          this.resetHighlightedSuggestion();
```

Comment:
If `highlightFirstSuggestion` is always false, this will reset the highlighted suggestion any time any other prop changes, which doesn't seem quite right.

I think it should only reset if all of the below are true:

* The new prop sets highlightFirstSuggestion to false
* The old prop had highlightFirstSuggestion set to true
* The first suggestion is currently highlighted

For example, if the first suggestion is highlighted, I manually select the 3rd item, and then `highlightFirstSuggestion` becomes false, I think it's probably better for the suggestion to stay where it was.

That should still work for your case, but ensures this behaviour is limited just to that specific issue, and doesn't affect normal suggestion selection elsewhere.

Reply:
Thanks for the detailed feedback. I agree and have updated the code now.